jvb/src/main/kotlin/org/jitsi/videobridge/load_management/JvbLoadManager.kt

```diff
@@ -39,17 +39,34 @@ class JvbLoadManager<T : JvbLoadMeasurement> @JvmOverloads constructor(
 
     fun loadUpdate(loadMeasurement: T) {
         logger.cdebug { "Got a load measurement of $loadMeasurement" }
+        val now = clock.instant()
         if (loadMeasurement.getLoad() >= jvbLoadThreshold.getLoad()) {
             state = State.OVERLOADED
-            logger.info("Load measurement $loadMeasurement is above threshold of $jvbLoadThreshold, " +
-                    "maybe running load reducer")
-            maybeRun("load reducer") { reduceLoad() }
+            logger.info("Load measurement $loadMeasurement is above threshold of $jvbLoadThreshold")
+            if (canRunReducer(now)) {
+                logger.info("Running load reducer")
+                loadReducer.reduceLoad()
+                lastReducerTime = now
+            } else {
+                logger.info("Load reducer ran at $lastReducerTime, which is within ${loadReducer.impactTime()} " +
+                    "of now, not running reduce")
+            }
         } else {
             state = State.NOT_OVERLOADED
             if (loadMeasurement.getLoad() < jvbRecoveryThreshold.getLoad()) {
-                logger.info("Load measurement $loadMeasurement is below threshold of $jvbLoadThreshold, " +
-                        "maybe running recovery")
-                maybeRun("recovery") { recover() }
+                if (canRunReducer(now)) {
+                    if (loadReducer.recover()) {
+                        logger.info("Recovery ran after a load measurement of $loadMeasurement (which was below " +
+                            "threshold of $jvbLoadThreshold) was received")
+                        lastReducerTime = now
```

Comment:
Note that I only set `lastReducerTime` in the `recover` case when there was actually work done: otherwise all normal measurements will register a run of the load reducer and we would have to wait longer to run `reduceLoad`.